File: 74. Search a 2D Matrix/search-a-2d-matrix.js
// 74. Search a 2D Matrix (7/1/53723)
// Runtime: 95 ms (21.10%) Memory: 39.58 MB (94.37%) 

/**
 * @param {number[][]} matrix
 * @param {number} target
 * @return {boolean}
 */
function searchMatrix(matrix, target) {
    for(let i = 0; i < matrix.length; i++) {
        for(let j = 0; j < matrix[i].length; j++) {
            if(matrix[i][j] === target) return true;
            if(matrix[i][j] > target) return false;
        }
    }
    return false;
};

File: 74. Search a 2D Matrix/search-a-2d-matrix.c
// 74. Search a 2D Matrix (4/13/54212)
// Runtime: 0 ms (92.69%) Memory: 6.34 MB (17.31%) 

bool searchMatrix (int** matrix, int matrix_size, int* matrix_col_size, int target) {
    int row = 0;
    for (int i = 0; i < matrix_size; i++) {
        if (matrix[i][0] <= target) {
            row = i;
        }
    }
    
    for (int i = 0; i < *matrix_col_size; i++) {
        if (matrix[row][i] == target) {
            return true;
        }
    }
    
    return false;
}
